runtime/parser/statement.go

```diff
@@ -96,31 +99,41 @@ func parseStatement(p *parser) (ast.Statement, error) {
 			return parseEmitStatement(p)
 		case keywordView:
 			purityPos := p.current.StartPos
+			// look ahead 1 token for the `fun` keyword
+			p.startBuffering()
```

Comment:
Given that we're not parsing a complex rule, but just want a couple tokens lookahead, you can simplify to saving and restoring the token stream cursor and current token:

```suggestion
			current := p.current
			cursor := p.tokens.Cursor()
```

This should be a bit more efficient than buffering.